lib/active_record/connection_adapters/sqlserver/core_ext/array_handler.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module ActiveRecord
+  module ConnectionAdapters
+    module SQLServer
+      module CoreExt
+        module ArrayHandler
+          module NullPredicate # :nodoc:
+            def self.or(other)
+              other
+            end
+          end
+
+          def call(attribute, value)
+            return attribute.in([]) if value.empty?
+
+            values = value.map { |x| x.is_a?(Base) ? x.id : x }
+            nils = values.extract!(&:nil?)
+            ranges = values.extract! { |v| v.is_a?(Range) }
+
+            values_predicate =
+              case values.length
+              when 0 then NullPredicate
+              when 1 then predicate_builder.build(attribute, values.first)
+              else
+                values.map! do |v|
+                  predicate_builder.build_bind_attribute(attribute.name, v)
+                end
+                values.empty? ? NullPredicate : attribute.in(values)
+              end
+
+            unless nils.empty?
+              values_predicate = values_predicate.or(predicate_builder.build(attribute, nil))
+            end
+
+            array_predicates = ranges.map { |range| predicate_builder.build(attribute, range) }
+            array_predicates.unshift(values_predicate)
+            array_predicates.inject(&:or)
+          end
```

Comment:
Can you explain the reasoning behind these changes? If possible, can you open a separate PR for each issue you are trying to address? It's easier for us to review and keep track of changes in future bug reports.

Reply:
This reverts an optimization made in rails 6.1, back to the functionality of 6.0. With the optimization present, any operations using an array would fail. 

For example, `where(id: [1,2,3])` would throw. 

I believe this was the commit that introduced the change: https://github.com/rails/rails/commit/72fd0bae5948c1169411941aeea6fef4c58f34a9

Sadly I don't remember the exact chain of events that lead me to this solution, it's been awhile.